### PR TITLE
Specs for changeset functions

### PR DESCRIPTION
**Pull request by alco**

### CODE REVIEW

File: lib/ecto/changeset.ex [hunk 1]
```diff
@@ -1419,6 +1422,7 @@ defmodule Ecto.Changeset do
       |> unique_constraint(:email)
 
   """
+  @spec unique_constraint(t, atom, options) :: t
```
Comment (michalmuskala) [inline lib/ecto/changeset.ex]: Why use in most places `Keyword.t` and in others `options` for the options?

Comment (alco) [inline lib/ecto/changeset.ex]: @michalmuskala I added those specs according to the usage of `options` in the functions. Those with the type `options` rely on Access, so maps would also work. Where it says `Keyword.t`, the function uses `Keyword` for fetching the options.

Comment (josevalim) [inline lib/ecto/changeset.ex]: Please use `Keyword.t` throughout. Using access is an implementation detail mostly, does not imply we want to support all access based types. ;)

**Comment by josevalim** [comment]
Thank you! I have added just one tiny comment!

**Comment by alco** [comment]
Updated.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**